src/hooks/use-pivot-table.ts

```diff
@@ -0,0 +1,121 @@
+import { useEffect, useState, type stardust } from "@nebula.js/stardust";
+import type { Root } from "react-dom/client";
+import render from "../pivot-table/Root";
+import { MAX_ROW_COUNT } from "../pivot-table/constants";
+import { usePVData } from "../pivot-table/data/use-pv-data";
+import type { Model } from "../types/QIX";
+import type { ExtendedSelections, ExtendedTranslator, LayoutService, StyleService, ViewService } from "../types/types";
+
+interface UsePivotTableProps {
+  model: Model;
+  rect: stardust.Rect;
+  constraints: stardust.Constraints;
+  selections: ExtendedSelections;
+  viewService: ViewService;
+  layoutService: LayoutService;
+  isLoading: boolean;
+  qPivotDataPages: EngineAPI.INxPivotPage[];
+  styleService: StyleService;
+  reactRoot: Root;
+  isFontLoaded: boolean;
+  translator: ExtendedTranslator;
+  language: string;
+}
+
+export interface PageInfo {
+  currentPage: number;
+  shouldShowPagination: boolean;
+  totalPages: number;
+  rowsPerPage: number;
+  totalRowCount: number;
+}
+
+const usePivotTable = ({
+  model,
+  rect,
+  constraints,
+  selections,
+  viewService,
+  layoutService,
+  isLoading,
+  qPivotDataPages,
+  styleService,
+  reactRoot,
+  isFontLoaded,
+  translator,
+  language,
+}: UsePivotTableProps) => {
+  const {
+    size,
+    layout: {
+      qHyperCube: { qSize },
+    },
+  } = layoutService;
+
+  const rowsPerPage = qSize.qcy <= MAX_ROW_COUNT ? qSize.qcy : MAX_ROW_COUNT;
```

Comment:
You should be able to do:
```suggestion
  const rowsPerPage = Math.min(qSize.qcy, MAX_ROW_COUNT):
```